src/devices/Board/Board.cs

```diff
@@ -0,0 +1,692 @@
+        private const string BaseBoardProductRegistryValue = @"SYSTEM\HardwareConfig\Current\BaseBoardProduct";
+        private const string RaspberryPi2Product = "Raspberry Pi 2";
+        private const string RaspberryPi3Product = "Raspberry Pi 3";
+        private const string HummingBoardProduct = "HummingBoard-Edge";
+
+        private readonly PinNumberingScheme _defaultNumberingScheme;
+        private readonly object _pinReservationsLock;
+        private readonly Dictionary<int, List<PinReservation>> _pinReservations;
+        private readonly Dictionary<int, I2cBusManager> _i2cBuses;
+        private bool _initialized;
+        private bool _disposed;
+
+        /// <summary>
+        /// Constructs a board instance with the given default numbering scheme.
+        /// All methods will use the given numbering scheme, unless another scheme is explicitly given in a call.
+        /// </summary>
+        /// <param name="defaultNumberingScheme">Default numbering scheme for the board.</param>
+        /// <remarks>
+        /// The constructor will never throw an exception. Call <see cref="Initialize"/> to initialize the hardware and check
+        /// whether an instance can actually run on the current hardware.
+        /// </remarks>
+        protected Board(PinNumberingScheme defaultNumberingScheme)
+        {
+            _defaultNumberingScheme = defaultNumberingScheme;
+            _pinReservations = new Dictionary<int, List<PinReservation>>();
+            _i2cBuses = new Dictionary<int, I2cBusManager>();
+            _pinReservationsLock = new object();
+            _initialized = false;
+            _disposed = false;
+        }
+
+        /// <summary>
+        /// True if the board instance is initialized
+        /// </summary>
+        protected bool Initialized => _initialized;
+
+        /// <summary>
+        /// True if this instance is disposed.
+        /// Any attempt to use it after this becomes true results in undefined behavior.
+        /// </summary>
+        protected bool Disposed
+        {
+            get
+            {
+                return _disposed;
+            }
+        }
+
+        /// <summary>
+        /// The default pin numbering scheme
+        /// </summary>
+        public PinNumberingScheme DefaultPinNumberingScheme
+        {
+            get
+            {
+                return _defaultNumberingScheme;
+            }
+        }
+
+        /// <summary>
+        /// Converts pin numbers from one numbering scheme to another.
+        /// </summary>
+        /// <param name="pinNumber">Pin number to convert</param>
+        /// <param name="inputScheme">The numbering scheme of the input pin</param>
+        /// <param name="outputScheme">The desired numbering scheme</param>
+        /// <returns>The converted pin number</returns>
+        public abstract int ConvertPinNumber(int pinNumber, PinNumberingScheme inputScheme, PinNumberingScheme outputScheme);
```

Comment:
I think I agree with Krzysztof here, we've learned that using PinNumberingScheme is more confusing that not, so I would just take an optional parameter in the Create factory method, and just assume that all ints are in that numbering scheme going forward.

Reply:
With the removal of the numbering scheme support, this went away, too.